tests/stdlib/tstrmisc.nim

```diff
@@ -0,0 +1,21 @@
+import strmisc
+
+
+func main() =
+  doAssert parseFloatThousandSep("0") == 0.0
+  doAssert parseFloatThousandSep("-0") == -0.0
+  doAssert parseFloatThousandSep("0.0") == 0.0
+  doAssert parseFloatThousandSep("1.0") == 1.0
+  doAssert parseFloatThousandSep("-0.0") == -0.0
+  doAssert parseFloatThousandSep("-1.0") == -1.0
+  doAssert parseFloatThousandSep("1.000") == 1.0
+  doAssert parseFloatThousandSep("-1.000") == -1.0
+  doAssertRaises(ValueError): discard parseFloatThousandSep("1,0000.000")
+  doAssertRaises(ValueError): discard parseFloatThousandSep("--")
+  doAssertRaises(ValueError): discard parseFloatThousandSep("..")
+  doAssertRaises(ValueError): discard parseFloatThousandSep("000,1.000,,,,,,,,,,,,,,000,,,,,0000")
```

Comment:
always show the "least extreme" failure modes (ie at the boundary) otherwise this weakens the test;
this is not a great example because it has multiple errors; instead maybe:
`parseFloatThousandSep("1,,000")`